scikeras/utils/random_state.py

```diff
@@ -1,14 +1,44 @@
 import os
 import random
 from contextlib import contextmanager
-from typing import Generator
+from typing import Generator, Iterator
 
 import numpy as np
-import tensorflow as tf
-from tensorflow.python.eager import context
-from tensorflow.python.framework import config, ops
 
-DIGITS = frozenset(str(i) for i in range(10))
+try:
+    import tensorflow as tf
+    from tensorflow.python.eager import context
+    from tensorflow.python.framework import config, ops
+
+    def tf_set_seed(seed: int) -> None:
```

Comment:
These definitions should be in an "else" clause

Reply:
You can write the following. It's good practice to keep the "try" part to a minimum so that unexpected exceptions aren't accidentally caught.

```python
try:
   ...
except ImportError:
    ...
else:
   # Only runs if there are no exceptions above
   ...
```

See the end of section 8.3 here: https://docs.python.org/3/tutorial/errors.html#handling-exceptions